src/components/vm/disks/diskAdd.tsx

```diff
@@ -391,10 +523,13 @@ export const AddDiskModalBody = ({ disk, idPrefix, isMediaInsertion, vm, vms, su
     }), [defaultPool, mode, vm]);
     const storagePoolName = diskParams.storagePool?.name;
 
-    const getPoolFormatAndDevice = (pool, volName) => {
-        const params = { format: getDefaultVolumeFormat(pool), device: "disk" };
+    const getPoolFormatAndDevice = (pool: StoragePool, volName: optString | false) => {
+        const params: Pick<DialogValues, "format" | "device"> = {
+            format: getDefaultVolumeFormat(pool),
+            device: "disk"
+        };
         if (volName && ['dir', 'fs', 'netfs', 'gluster', 'vstorage'].indexOf(pool.type) > -1) {
-            const volume = pool.volumes.find(vol => vol.name === volName);
+            const volume = (pool.volumes || []).find(vol => vol.name === volName);
```

Comment:
Ah, that's because the (D-Bus?) API isn't declared strict enough to point out that `volumes` is always defined? If that wasn't so, the code would have crashed before. Could this become a `!` to clarify that, instead of adding a runtime impact and weakening the expected API?

Reply:
I took the big hammer and added the "storagepools: Make sure pool.volumes is always an array" commit.